java-checks/src/main/java/org/sonar/java/filters/JavaIssueFilter.java

```diff
@@ -22,14 +22,28 @@
 import org.sonar.api.issue.Issue;
 import org.sonar.plugins.java.api.JavaCheck;
 import org.sonar.plugins.java.api.JavaFileScanner;
+import org.sonar.plugins.java.api.tree.Tree;
 
 import java.util.Set;
 
 public interface JavaIssueFilter extends JavaFileScanner {
 
-  void setComponentKey(String componentKey);
+  enum IssueState {
+    ACCEPTED, REFUSED;
 
-  boolean accept(Issue issue);
+    public static IssueState shouldBeRefused(boolean shouldBeRefused) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Provide multiple methods instead of using "shouldBeRefused" to determine which action to take. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2301)